src/Docker/Client/Http.hs

```diff
@@ -94,7 +94,7 @@ runDockerT (opts, h) r = runReaderT (unDockerT r) (opts, h)
 mkHttpRequest :: HttpVerb -> Endpoint -> DockerClientOpts -> Maybe Request
 mkHttpRequest verb e opts = request
         where fullE = T.unpack (baseUrl opts) ++ T.unpack (getEndpoint (apiVer opts) e)
-              initialR = parseUrl fullE
+              initialR = parseUrlThrow fullE
```

Comment:
Do we want `parseRequest` instead? According to the docs, [parseUrlThrow](https://www.stackage.org/haddock/lts-8.5/http-client-0.5.6.1/Network-HTTP-Client.html#v:parseUrlThrow) throws an exception on non-2XX, but we already handle this with `statusCodeToError`.